src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/properties/potions/CustomPotionEffectsFluidProperty.java

```diff
@@ -0,0 +1,44 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.properties.potions;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.properties.FluidProperty;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import it.unimi.dsi.fastutil.ints.IntSet;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+
+public class CustomPotionEffectsFluidProperty implements FluidProperty<ListTag> {
+	@Override
+	public ListTag merge(Fluid fluidA, Fraction amountA, Fraction amountB, ListTag a, ListTag b) {
+		ListTag tags = a.copy();
+
+		for (Tag tag : a) {
+			tags.add(tag.copy());
+		}
+
+		return tags;
+	}
+
+	@Override
+	public boolean areCompatible(Fluid fluidA, ListTag a, ListTag b) {
+		IntSet set = new IntOpenHashSet();
+
+		for (Tag tag : a) {
+			CompoundTag effect = (CompoundTag) tag;
+			set.add(effect.getInt("Id"));
```

Comment:
I'd say use the identifier rather than the raw id. Raw ids are not guaranteed to be the same on every instance

Reply:
that's a minecraft thing https://minecraft.gamepedia.com/Potion

Reply:
So this refers to the item?